src/main/java/software/amazon/awssdk/crt/http/Http2StreamManagerOptions.java

```diff
@@ -129,29 +139,130 @@ public HttpClientConnectionManagerOptions getConnectionManagerOptions() {
+    public Http2StreamManagerOptions withConnectionManagerOptions(
+            HttpClientConnectionManagerOptions connectionManagerOptions) {
         this.connectionManagerOptions = connectionManagerOptions;
         return this;
     }
 
     /**
-     * Validate the stream manager options are valid to use. Throw exceptions if not.
+     * @return Prior knowledge is used or not
+     */
+    public boolean isPriorKnowledge() {
+        return priorKnowledge;
+    }
+
+    /**
+     * Set to true to use prior knowledge to setup connection.
+     * If any TLS was set, exception will be raised if prior knowledge is set during
+     * validation.
+     * If NO TLS was set, exception will be raised if prior knowledge is NOT set
+     * during validation.
+     *
+     * @param priorKnowledge Prior knowledge used or not.
+     * @return this
+     */
+    public Http2StreamManagerOptions withPriorKnowledge(boolean priorKnowledge) {
+        this.priorKnowledge = priorKnowledge;
+        return this;
+    }
+
+    /**
+     * @return Connection closed or not when server error happened (500/502/503/504
+     *         response status code received).
+     */
+    public boolean isCloseConnectionOnServerError() {
+        return closeConnectionOnServerError;
+    }
+
+    /**
+     * Set to true to inform stream manager to close connection when response with
+     * 500/502/503/504 received.
+     * Stream manager will stop using the connection with server error will start a
+     * new connection for other streams.
+     *
+     * @param closeConnectionOnServerError Connection closed or not when server
+     *                                     error happened.
+     * @return this
+     */
+    public Http2StreamManagerOptions withCloseConnectionOnServerError(boolean closeConnectionOnServerError) {
+        this.closeConnectionOnServerError = closeConnectionOnServerError;
+        return this;
+    }
+
+    /**
+     * Settings to control the period ping to be sent for connections held by stream
+     * manager.
+     *
+     * @param connectionPingPeriodMs  The period for all the connections held by
+     *                                stream manager to send a PING in milliseconds.
+     *                                If you specify 0, manager will NOT send any
+     *                                PING.
+     * @param connectionPingTimeoutMs Network connection will be closed if a ping
+     *                                response is not received
+     *                                within this amount of time (milliseconds).
+     *                                If you specify 0, a default value will be
+     *                                used. If this is larger than
+     *                                connectionPingPeriodMs,
+     *                                it will be capped to be equal.
+     * @return this
+     */
+    public Http2StreamManagerOptions withConnectionPeriodPingSettings(int connectionPingPeriodMs,
```

Comment:
trivial: since this sets 2 variables, maybe just shorten to `withConnectionPing(int periodMs, int timeoutMs)`

or have 2 separate setters `withConnectionPingPeriodMs(int periodMs)` and `withConnectionPingTimeoutMs(int timeoutMs)`